### PR TITLE
Standardize string-attribute-conversion logic.

### PR DESCRIPTION
**Pull request by diegoreymendez**
## Details:

It propagates the same changes across different attribute converters (`em` and `u`).

It also adds unit tests for the case of `em` and `i`.  Running those tests in `develop` should fail.

## Dependencies:

This PR builds up on the fix in https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1138.

### Test 1:

Run the provided tests in develop and make sure the first one fails.
Run the tests in this PR and make sure none fail.

### Test 2:

1. Open the empty editor.
2. Switch to HTML mode.
3. Type `<i>Hello!</i>`.
4. Switch to visual mode and back to HTML mode.

Make sure the final conversion doesn't have an extra `em` tag.

### Test 3:

1. Open the empty editor.
2. Switch to HTML mode.
3. Type `<b>Hello!</b>`.
4. Switch to visual mode and back to HTML mode.

Make sure the final conversion doesn't have an extra `strong` tag.

### Test 4:

1. Open the empty editor.
2. Switch to HTML mode.
3. Type `<u>Hello!</u>`.
4. Switch to visual mode and back to HTML mode.

Make sure the output conversion looks good.

### CODE REVIEW

**Comment by SergioEstevao** [comment]
I just noticed an issue that I think it's non-related to this changes:

When you move around the cursor it looks the toolbar gets updated with the previous position of the cursor (some kind of delay).

**Comment by diegoreymendez** [comment]
> When you move around the cursor it looks the toolbar gets updated with the previous position of the cursor (some kind of delay).

For context: as discussed via DM, I can repro the same issue in `develop` so I'll open an issue to tackle it separately and link it here.

**Comment by diegoreymendez** [comment]
@SergioEstevao - Issue created: https://github.com/wordpress-mobile/AztecEditor-iOS/issues/1140

File: Aztec/Classes/Converters/StringAttributesToAttributes/Utility/HTMLStyleToggler.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            elementNode.removeCSSAttributes(matching: cssAttributeMatcher)
+        }
+        
+        return elementNodes
+    }
+    
+    open func enable(in elementNodes: [ElementNode]) -> [ElementNode] {
+        var elementNodes = elementNodes
+        
+        // We can now check if we have any CSS attribute representing bold.  If that's the case we can completely skip
```
Comment (pinarol) [inline Aztec/Classes/Converters/StringAttributesToAttributes/Utility/HTMLStyleToggler.swift]: code comment still says bold

File: Aztec/Classes/Converters/StringAttributesToAttributes/Utility/HTMLStyleToggler.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        //
+        for elementNode in elementNodes {
+            let elementRepresentsStyle = defaultElement.equivalentNames.contains(elementNode.type)
+            
+            if elementRepresentsStyle || elementNode.containsCSSAttribute(matching: cssAttributeMatcher) {
+                return elementNodes
+            }
+        }
+        
+        // Nothing was found to represent bold... just add the element.
```
Comment (pinarol) [inline Aztec/Classes/Converters/StringAttributesToAttributes/Utility/HTMLStyleToggler.swift]: here also

**Comment by diegoreymendez** [comment]
Good observations @pinarol .  Thanks both for the review!

**Pull request merged**